Mark keys as optional

DirkBaumeister opened the pull request:
As suggested yesterday here is the pull request for optional keys per language. The intention is to mark specific keys as "only for one language" so that the translators see that no translation is needed.

@DSB you have to add a boolean field 'clf' to the keys table to make it functionable :)

Commit:
updated entries controller
updated language files
updated language entries model
updated views
updated styles

DSB commented:
We talked about a few things that should be changed. An update based on our discussion would be fine. If everything is fine this feature will make it into the core. 
Thanks.

DirkBaumeister committed:
updated entries controller to use helper table for optional translations
updated language files for optional translations
updated the moddle for language entries to match the helper table for optional translations
updated the views to correspond with the new helper table
added an database schema update file for the helper table
added css styles

DirkBaumeister commented:
Hey,
I hope I fixed all the mentioned things we discussed two days ago :)
Additionally I also added an fix on the "getStatus" method: It now also recognizes if there is an translation entry for a key/language pair and its empty. Than it will not count to the stats on the index page.

Hope everything works fine now :)
